Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponAnimationController.cs

```diff
@@ -34,25 +39,12 @@ public class WeaponAnimationController : MonoBehaviour, IWeaponAnimationProvider
         [Resolve][SerializeField] private GameObject _weaponHolder = null;
         [Resolve][SerializeField] private Animator _weaponAnimator = null;
         private Camera _mainCam;
-        
-        private Func<float> _characterVelocity = null;
-        private Func<bool> _isWallRunning = null;
-        private Func<bool> _isSliding = null;
-        private Func<bool> _isDashing = null;
-        private Func<bool> _isGrounded = null;
 
-        public void Initialize(
-            Func<float> characterVelocity, 
-            Func<bool> isWallRunning,
-            Func<bool> isGrounded,
-            Func<bool> isSliding,
-            Func<bool> isDashing)
+        private IWeaponBobIntensity _weaponBobIntensity;
```

Comment:
Reminder - interfaces is NOT supposed to expose variables, it IS supposed to provide methods/logic to what needs it.